store/src/state.rs

```diff
@@ -342,27 +342,27 @@ impl State {
         account_ids: &[AccountId],
         note_tag_prefixes: &[u32],
         nullifier_prefixes: &[u32],
-    ) -> Result<(StateSyncUpdate, MmrDelta, MerklePath), anyhow::Error> {
+    ) -> Result<(StateSyncUpdate, MmrDelta), anyhow::Error> {
         let inner = self.inner.read().await;
 
         let state_sync = self
             .db
             .get_state_sync(block_num, account_ids, note_tag_prefixes, nullifier_prefixes)
             .await?;
 
-        let (delta, path) = {
-            let delta = inner
-                .chain_mmr
-                .get_delta(block_num as usize, state_sync.block_header.block_num as usize)?;
-
-            let proof = inner
-                .chain_mmr
-                .open(block_num as usize, state_sync.block_header.block_num as usize)?;
-
-            (delta, proof.merkle_path)
-        };
-
-        Ok((state_sync, delta, path))
+        // Important notes about the boundary conditions:
+        //
+        // - The Mmr forest is 1-indexed whereas the block number is 0-indexed. The Mmr root
+        // contained in the block header always lag behind by one block, this is because the Mmr
+        // leaves are hashes of block headers, and we can't have self-referential hashes. These two
+        // points cancel out and don't require adjusting.
+        // - Mmr::get_delta is inclusive, whereas the sync_state request block_num is defined to be
+        // exclusive, so the from_forest has to be adjusted with a +1
+        let delta = inner
+            .chain_mmr
+            .get_delta((block_num + 1) as usize, state_sync.block_header.block_num as usize)?;
```

Comment:
Wouldn't this fail if `block_num + 1 > state_sync.block_header.block_num` (because of [this](https://github.com/0xPolygonMiden/crypto/blob/next/src/merkle/mmr/full.rs#L174) condition)?

This could happen if both client and node are at block $n$, and the client makes a request using $n$ as `block_num`.

Reply:
ah, good point, added code to handle the corner case.